packages/typespec-client-generator-core/src/types.ts

```diff
@@ -136,7 +136,15 @@ function getAnyType(context: TCGCContext, type: Type): [SdkBuiltInType, readonly
 
 function getEncodeHelper(context: TCGCContext, type: Type, kind: string): string {
   if (type.kind === "ModelProperty" || type.kind === "Scalar") {
-    return getEncode(context.program, type)?.encoding || kind;
+    const encode = getEncode(context.program, type);
+    if (encode?.encoding) {
+      return encode.encoding;
+    }
+    if (encode?.type) {
+      // if we specify the encoding type in the decorator, we set the `.encode` string
+      // to the kind of the encoding type
+      return getSdkBuiltInType(context, encode.type).kind;
+    }
```

Comment:
Should we add a test for this case?

I am not sure whether there is new update on `@encode` to dev libs.

At current 0.58, the decorator is
```
extern dec encode(
  target: Scalar | ModelProperty,
  encoding: string | EnumMember,
  encodedAs?: Scalar
);
```

And `@encode(string)` works as it matches this part `encoding: string | EnumMember,`

Reply:
they haven't added support yet for `@encode(string)` yet into main, so there's no way to test it right now. The changes here are pre-emptive, so once that change goes in, we'll automatically support `@encode(string)`. I've opened an [issue](https://github.com/Azure/typespec-azure/issues/1264) to add a test once it's in on the compiler side